modules/CompilerOptions.cmake

```diff
@@ -0,0 +1,234 @@
+                message(WARNING ${error_msg})
+            endif(ENABLE_ECC)
+        endif()
+
+        # acx_openssl_gost.m4
+        set(testfile ${CMAKE_SOURCE_DIR}/modules/tests/test_openssl_gost.c)
+        try_run(RUN_GOST COMPILE_RESULT
+                "${CMAKE_BINARY_DIR}/prebuild_santity_tests" ${testfile}
+                LINK_LIBRARIES ${CRYPTO_LIBS}
+                CMAKE_FLAGS
+                    "-DINCLUDE_DIRECTORIES=${CRYPTO_INCLUDES}"
+                )
+        if(NOT COMPILE_RESULT)
+            message(FATAL_ERROR "failed to compile: ${testfile}")
+        endif(NOT COMPILE_RESULT)
+
+        if(RUN_GOST EQUAL 0)
+            message(STATUS "OpenSSL: found GOST engine")
+        else()
+            set(error_msg "OpenSSL: cannot find GOST engine! OpenSSL library has no GOST support!")
+            if(ENABLE_GOST)
+                message(FATAL_ERROR ${error_msg})
+            else(ENABLE_GOST)
+                message(WARNING ${error_msg})
+            endif(ENABLE_GOST)
+        endif()
+
+        # acx_openssl_fips.m4
+        set(testfile ${CMAKE_SOURCE_DIR}/modules/tests/test_openssl_fips.c)
+        try_run(RUN_FIPS COMPILE_RESULT
+                "${CMAKE_BINARY_DIR}/prebuild_santity_tests" ${testfile}
+                LINK_LIBRARIES ${CRYPTO_LIBS}
+                CMAKE_FLAGS
+                    "-DINCLUDE_DIRECTORIES=${CRYPTO_INCLUDES}"
+                )
+        if(NOT COMPILE_RESULT)
+            message(FATAL_ERROR "failed to compile: ${testfile}")
+        endif(NOT COMPILE_RESULT)
+
+        if(RUN_FIPS EQUAL 0)
+            message(STATUS "OpenSSL: found working FIPS_mode_set()")
+        else()
+            set(error_msg "OpenSSL: FIPS_mode_set(1) failed. OpenSSL library is not FIPS capable!")
+            if(ENABLE_FIPS)
+                message(FATAL_ERROR ${error_msg})
+            else(ENABLE_FIPS)
+                message(WARNING ${error_msg})
+            endif(ENABLE_FIPS)
+        endif()
+
+        # acx_openssl_rfc3349
+        # acx_openssl_rfc5649
+
+    else(OPENSSL_FOUND)
+        message(FATAL_ERROR "Failed to find OpenSSL!")
+    endif(OPENSSL_FOUND)
+endif(WITH_OPENSSL)
+
+# Find SQLite3
+if(WITH_SQLITE3)
+    include(FindSQLite3)
+    if(SQLITE3_FOUND)
+        message(STATUS "Found SQLite3")
+        set(SQLITE3_INCLUDES ${SQLITE3_INCLUDE_DIRS})
+        set(SQLITE3_LIBS ${SQLITE3_LIBRARIES})
+    else(SQLITE3_FOUND)
+        message(FATAL_ERROR "Failed to find SQLite3!")
+    endif(SQLITE3_FOUND)
+endif(WITH_SQLITE3)
+
+if(BUILD_TESTS)
+    # Find CppUnit (equivalent of acx_cppunit.m4)
+    include(FindCppUnit)
+    if(CPPUNIT_FOUND)
+        message(STATUS "Found CppUnit")
+        set(CPPUNIT_INCLUDES ${CPPUNIT_INCLUDE_DIR})
+        set(CPPUNIT_LIBS ${CPPUNIT_LIBRARY})
+    else(CPPUNIT_FOUND)
+        message(FATAL_ERROR "Failed to find CppUnit!")
+    endif(CPPUNIT_FOUND)
+endif(BUILD_TESTS)
+
+configure_file(config.h.in.cmake ${CMAKE_BINARY_DIR}/config.h)
```

Comment:
This file does not seem to exist.  Changing to config.h.in seems to work

Reply:
git for some reason didn't detect it, so I had to force add it.